### PR TITLE
Core/Misc: utf8 improvments, prevent possible money dupe in mailhandler and also change damage to float

### PR DESCRIPTION
**Pull request by Viste**
https://youtu.be/ttJBdr6eBuo
## CHANGES PROPOSED:
-  Add utf8limit helper function for truncating UTF8 strings by bytes count without corruption(Warlockbugs )
-  cleanup in utf8length
-  limit NormalizePlayerName by  MAX_INTERNAL_PLAYER_NAME
- other cleanups in utf8handle code
-  prevent possible money dupe in mailhandler
- change damage var to float

## HOW TO TEST THE CHANGES:
try to use limiter in code...   
test mails and damage

## Target branch(es):
- [x] Master

### CODE REVIEW

File: src/common/Utilities/Util.cpp [hunk 1]
```diff
@@ -541,19 +544,26 @@ void utf8printf(FILE* out, const char *str, ...)
...
+    std::size_t temp_len = vsnprintf(&temp_buf[0], 32 * 1024, str, *ap);
+    temp_buf.resize(strnlen_s(temp_buf.c_str())); // Resize to match the formatted string
 
-    size_t wtemp_len = 32*1024-1;
-    Utf8toWStr(temp_buf, temp_len, wtemp_buf, wtemp_len);
+    if (!temp_buf.empty())
+    {
+        Utf8toWStr(temp_buf, wtemp_buf, 32 * 1024);
+        wtemp_buf.push_back('\0');
+
```
Comment (pklloveyou) [inline src/common/Utilities/Util.cpp]: The lack of "} "?

Comment (pklloveyou) [inline src/common/Utilities/Util.cpp]: There might be a parenthesis missing here，

Comment (Viste) [reply src/common/Utilities/Util.cpp]: ???

Comment (Viste) [reply src/common/Utilities/Util.cpp]: ah i see lol )

Comment (Viste) [reply src/common/Utilities/Util.cpp]: but this is not need for #2748 you need only src/server/game/Handlers/MailHandler.cpp changes

**Comment by BarbzYHOOL** [comment]
Seems like @pklloveyou tested the PR for a few days here https://github.com/azerothcore/azerothcore-wotlk/issues/2748

It's not 100% proven but seems promising!

**Comment by pklloveyou** [comment]
There should be an error if the cheater shows up，I'm still waiting for the cheater to show up

**Comment by FrancescoBorzi** [comment]
@pklloveyou if you've been using this and can ensure us that no functionality is broken, we can merge this already

**Comment by pklloveyou** [comment]
I only merged that part of the mail.。

**Comment by Viste** [comment]
hm i can separate PR all utf8 shit split in one(with bytebuffer cleanup) and mail in another

**Comment by FrancescoBorzi** [comment]
yes please, smaller PRs are always better

**Comment by pklloveyou** [comment]
I tried to merge the remaining parts, there were two errors
image
image
image

**Comment by Viste** [comment]
> I tried to merge the remaining parts, there were two errors
> image

yes

**Comment by pklloveyou** [comment]
Thank you very much for your answer.

**Comment by pklloveyou** [comment]
Thank you very much for your answer.

**Comment by Viste** [comment]
can you show errors?

**Comment by pklloveyou** [comment]
I updated the latest version of VS, I will try again

**Comment by pklloveyou** [comment]
VS could not be built after the update. I will fix my build problem first

**Comment by Viste** [comment]
part of this will be moved https://github.com/azerothcore/azerothcore-wotlk/pull/2767 and other  in new

**Pull request closed without merging**